pkg/requirements/requirements.go

```diff
@@ -48,3 +38,59 @@ func CurrentRequirements(tmpDir string) (string, error) {
 	}
 	return requirementsFile, nil
 }
+
+func ReadRequirements(path string) ([]string, error) {
+	fh, err := os.Open(path)
+	if err != nil {
+		return nil, err
+	}
+	// Use scanner to handle CRLF endings
+	scanner := bufio.NewScanner(fh)
+	scanner.Split(scanLinesWithContinuations)
+	requirements := []string{}
+	for scanner.Scan() {
+		requirementsText := strings.TrimSpace(scanner.Text())
+		if len(requirementsText) == 0 {
+			continue
+		}
```

Comment:
Not a big deal but as we're implementing a custom splitter we could have `scanLinesWithContinuations` return `nil` for any whitespace only and comment lines and then they'll be skipped here entirely.

Reply:
Resolved in [d4397ba](https://github.com/replicate/cog/pull/2230/commits/d4397bafd8e53a37f13ea408fc66d359991a4766)

Reply:
I revert this in [36131fa](https://github.com/replicate/cog/pull/2230/commits/36131fa1b5839b8c73029621f566d76a96623bc9). It seems that if you return nil as a token the scanner stops scanning in the loop.